go/billing/tasks.py

```diff
@@ -357,6 +357,36 @@ def generate_chunks(item_iter, items_per_chunk=1000, sep="\n"):
     return archive
 
 
+@task
+def gen_statement_then_archive(account_id, from_date, to_date, delete=True):
+    """Task to generate a monthly statement for the given account,
+    then archive the account.
+    """
+    generate_monthly_statement(account_id, from_date, to_date)
+    archive_transactions(account_id, from_date, to_date, delete)
+
+
+@task()
+def gen_statement_then_archive_monthly(months_ago=1):
+    """Spawn sub-tasks to generate a monthly statement
+    then archive each account.
+    """
+    from_date, to_date = month_range(months_ago=months_ago)
+
+    account_list = Account.objects.exclude(
+        statement__type=Statement.TYPE_MONTHLY,
+        statement__from_date=from_date,
+        statement__to_date=to_date)
+
+    account_list = account_list.exclude(
+        transactionarchive__from_date=from_date,
+        transactionarchive__to_date=to_date)
```

Comment:
The idea behind these exclusions is that they make re-running the task easier. With just one operation happening, that was a bit simpler. Now we have two operations and we exclude all accounts that have had either of them happen. I think that makes sense, and will mean we'll only have to manually re-run things for accounts that were midway through when something broken? Mostly just noting this here so we think about it.